circle.yml

```diff
@@ -0,0 +1,47 @@
+machine:
+  php:
+    version: 5.5.21
+  ruby:
+    version: 2.1.3
+  environment:
+    APP_ENV: testing
+
+general:
+  artifacts:
+    - "tests/codeception/_output"
+    - "storage/logs"
+
+dependencies:
+  cache_directories:
+    - elasticsearch-1.6.0
+    - node_modules
+    - public/components
+    - vendor
+  pre:
+    # - sed -i 's/^;//' ~/.phpenv/versions/$(phpenv global)/etc/conf.d/xdebug.ini
+    - echo "memory_limit = 1024M" > ~/.phpenv/versions/$(phpenv global)/etc/conf.d/memory.ini
+    - if [[ ! -e elasticsearch-1.6.0 ]]; then wget https://download.elastic.co/elasticsearch/elasticsearch/elasticsearch-1.6.0.tar.gz && tar -xvf elasticsearch-1.6.0.tar.gz; fi
+    - elasticsearch-1.6.0/bin/elasticsearch: {background: true}
```

Comment:
While we can move the version to a variable (through `machine.environment`) you have to repeat it anyway in the `cached_directory` section as those are just strings and don't interpolate anything so wouldn't bring much benefits. Also seems to cause issues as far as I've tested, but not sure why.

Not reusing the app's ES provision is done on purpose as it contains a lot of things that are either uneeded for tests and/or specific to the Vagrant box. Also if we want to add ES plugins like Maven or something, those would just slow the build down for no benefits. We don't update ES that often anyway so, didn't think it was worth it